doc/connecting-lab.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Connecting LAVA Lab to the pipeline instance"
+date: 2024-02-19
+description: "Connecting a LAVA lab to the KernelCI pipeline"
+weight: 2
+---
+
+As we are moving towards the new KernelCI API and pipeline, we need to make sure
+all the existing LAVA labs are connected to the new pipeline instance.  This
+document explains how to do this.
+
+## Token setup
+
+The first step is to generate a token for the lab.  This is done by the lab admin,
+and the token is used to submit jobs from pipeline to the lab, and to authenticate
+LAVA lab callbacks to the pipeline.
+Requirements for the token:
+- Description: a string matching the regular expression `[a-zA-Z0-9\-]+`, for example "kernelci-new-api-callback"
+- Value: arbitrary, kept secret
+
+## Pipeline configuration
+
+### Secrets (toml) file
+
+The next step is to add the token to the pipeline services configuration files.
+Secrets are stored in the `kernelci.secrets` section of the `kernelci.toml` file,
+and added manually by the KernelCI system administrators.  For example, the token
+can be added to the `runtime.lava-labname` section of the `services.toml` file:
+
+```toml
+[runtime.lava-labname]
+runtime_token="TOKEN-VALUE"
+```
+
+### docker-compose file
+
+You need to add lab name to the scheduler-lava service in the `docker-compose.yml` file.
+
+### yaml pull request
+
+The final step is to submit a pull request to the `kernelci-pipeline` repository
+to add the lab configuration to the yaml file.
+For example, see the following [pull request](https://github.com/kernelci/kernelci-pipeline/pull/426).
+
+In details the pull request should add a new entry to the `runtimes` section of the configuration file:
+
+```yaml
+
+  lava-broonie:
+    lab_type: lava
+    url: 'https://lava.sirena.org.uk/'
+    priority_min: 10
+    priority_max: 40
+    notify:
+      callback:
+        token: kernelci-new-api-callback
+        url: https://staging.kernelci.org:9100
+
+```
+Where `lava-broonie` is the name of the lab, `lava` is the type of the lab, `url` is the URL of the lab, `priority_min` and `priority_max` are the priority range allowed to jobs, assigned by lab owner, and `notify` is the notification configuration for the lab.  The `callback` section contains the token and the URL of the pipeline instance LAVA callback endpoint.
```

Comment:
Instead of describing each field like this, the below section seems more readable to me:
```suggestion
  <lava-lab-name>:
    lab_type: lava
    url: '<url-to-the-lab>'
    priority_min: <min-priority-range>
    priority_max: <max-priority-range>
    notify:
      callback:
        token: <lab-token>
        url: <pipeline-lava-callback-url>
```
This can be followed by an example as you have already mentioned:
```
lava-broonie:
    lab_type: lava
    url: 'https://lava.sirena.org.uk/'
    priority_min: 10
    priority_max: 40
    notify:
      callback:
        token: kernelci-new-api-callback
        url: https://staging.kernelci.org:9100
```

Reply:
people get confused with brackets only example, they might add for example  `priority_min: <10>`, this is why i give real examples

Reply:
No, we should keep the example as it is just the way you have mentioned. I was suggesting to change the description of the example.